### PR TITLE
Add configuration option for zpool, usage of S3 and cron builds

### PR DESCRIPTION
**Pull request by tolbrino**
These changes make the role more useful for a steady use-case.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ S3 secret key. Default: `''`.
 
 ##### s3_bucket_name
 The bucket to use to store the packages and build options. Default: `''`.
+If no bucket is specified, no S3 related configuration will be performed.
```
Comment (JoergFiedler) [inline README.md]: I would introduce a own variable to enable/disable S3 syncronization…

File: tasks/main.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 - name: Install awscli
   pip: name=awscli state=present
+  when: s3_bucket_name != ''
```
Comment (JoergFiedler) [inline tasks/main.yml]: I would rather move all S3 tasks to a separate file …

File: tasks/main.yml [hunk 3]
```diff
@@ -44,6 +45,7 @@
            /usr/local/bin/aws s3 sync \
            s3://{{ s3_bucket_name }}{{ s3_upload_path }}/build-options/ \
            {{ poudriere_config_path }}/
+  when: s3_bucket_name != ''
```
Comment (JoergFiedler) [inline tasks/main.yml]: I would rather move all S3 tasks to a separate file …

File: tasks/main.yml [hunk 4]
```diff
@@ -57,8 +59,16 @@
   template: src=upload-to-s3.p2
             dest=/usr/local/bin/upload-to-s3
             mode=0750
+  when: s3_bucket_name != ''
```
Comment (JoergFiedler) [inline tasks/main.yml]: I would rather move all S3 tasks to a separate file …

File: tasks/main.yml [hunk 5]
```diff
@@ -57,8 +59,16 @@
...
 
 - name: Install build script
   template: src=build-ports.p2
             dest=/usr/local/bin/build-ports
             mode=0750
+
+- name: Enable daily build cron job
+  cron:
+    name: Run poudriere build
+    job: /usr/local/bin/build-ports
```
Comment (JoergFiedler) [inline tasks/main.yml]: param `-y` missing?

File: templates/build-ports.p2 [hunk 6]
```diff
@@ -20,6 +22,10 @@ _build_port()
 
 for jail in ${JAIL_NAMES}
 do
+  if [ "$1" == "-y" ]; then
```
Comment (JoergFiedler) [inline templates/build-ports.p2:25]: I would prefer `if` … `else` in favor of `continue`

File: templates/build-ports.p2 [hunk 7]
```diff
@@ -20,6 +22,10 @@ _build_port()
...
+      _build_port $jail
```
Comment (JoergFiedler) [inline templates/build-ports.p2:26]: I haven't tried it yet, so one question: when you skip `options` step, it means all packages are build using their default options, right?

**Review by JoergFiedler — COMMENTED** [review]
thank you very much for this idea … I am happy to merge this … if you want to make the changes yourself pls do so, otherwise I would most certainly apply them after I merged your PR …

File: templates/build-ports.p2 (hunk 7 above)
Comment (tolbrino) [reply templates/build-ports.p2:26]: Yes. This way one can run the options step manually at some time and they will be picked up when the build is executed next. The idea is that the builds can run unattended.

File: README.md (hunk 1 above)
Comment (tolbrino) [reply README.md]: I thought about that too. Will adapt it.

File: tasks/main.yml (hunk 2 above)
Comment (tolbrino) [reply tasks/main.yml]: Agreed.

File: tasks/main.yml (hunk 5 above)
Comment (tolbrino) [reply tasks/main.yml]: Obviously, thanks.

**Comment by tolbrino** [comment]
I will make the changes.

**Comment by JoergFiedler** [comment]
@tolbrino thank you so much …

**Pull request merged**